src/App.tsx

```diff
@@ -1,26 +1,163 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { wait } from './utils/fetchClient';
+import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { creatTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { getFilterTodos } from './utils/getFilterTodos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Default);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
```

Comment:
```suggestion
  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
```
You can store not only todos that you plan to delete.
As of now I don't see any loaders when I click the button to toggle all todos statuses